atlas/packages/components/src/components/Carousel/Carousel.tsx

```diff
@@ -1,39 +1,107 @@
-import React, { ReactNode, useRef } from "react";
+import React, { useState, useRef, useEffect, useCallback, useMemo } from "react";
+import { SerializedStyles } from "@emotion/core";
+import { animated, useSpring } from "react-spring";
+import useResizeObserver from "use-resize-observer";
 import { useCSS, CarouselStyleProps } from "./Carousel.style";
 import NavButton from "../NavButton";
 
 type CarouselProps = {
-	children: Array<ReactNode>;
-	scrollAmount?: Number;
+	children: React.ReactNode;
+	containerCss: SerializedStyles;
+	leftControlCss: SerializedStyles;
+	rightControlCss: SerializedStyles;
+	onScroll: (direction: "left" | "right") => void;
 } & CarouselStyleProps;
 
-export default function Carousel({ children, scrollAmount = 200, ...styleProps }: CarouselProps) {
-	const container = useRef(null);
+const Carousel: React.FC<Partial<CarouselProps>> = ({
+	children,
+	containerCss,
+	leftControlCss,
+	rightControlCss,
+	onScroll = () => {},
+}) => {
+	if (!Array.isArray(children)) {
+		return <>{children}</>;
```

Comment:
This will be visible better once we enforce eslint, but we shouldn't return here because we use hooks later on. Per hooks rules they shouldn't be conditional so we can return only after we've called all the hooks

Reply:
Moved it down and added a check to `useEffect` to prevent errors.